eks/modules/eks/worker-iam.tf

```diff
@@ -0,0 +1,36 @@
+
+data "aws_iam_policy_document" "worker-role-policy" {
+  statement {
+    actions = ["sts:AssumeRole"]
+
+    principals {
+      type        = "Service"
+      identifiers = ["ec2.amazonaws.com"]
+    }
+  }
+}
+resource "aws_iam_role" "aws-node" {
+  name = "terraform-eks-aws-node"
+
+  assume_role_policy = "${data.aws_iam_policy_document.worker-role-policy.json}"
+}
+
+resource "aws_iam_role_policy_attachment" "aws-node-AmazonEKSWorkerNodePolicy" {
+  policy_arn = "arn:aws:iam::aws:policy/AmazonEKSWorkerNodePolicy"
+  role       = "${aws_iam_role.aws-node.name}"
+}
+
+resource "aws_iam_role_policy_attachment" "aws-node-AmazonEKS_CNI_Policy" {
+  policy_arn = "arn:aws:iam::aws:policy/AmazonEKS_CNI_Policy"
+  role       = "${aws_iam_role.aws-node.name}"
+}
+
+resource "aws_iam_role_policy_attachment" "aws-node-AmazonEC2ContainerRegistryReadOnly" {
+  policy_arn = "arn:aws:iam::aws:policy/AmazonEC2ContainerRegistryReadOnly"
+  role       = "${aws_iam_role.aws-node.name}"
+}
+
+resource "aws_iam_instance_profile" "aws-node" {
+  name = "terraform-eks-aws"
```

Comment:
`role = "${aws_iam_role.aws-node.name}"`
needs to be set through variable interpolation otherwise, the deployment will be only possible for one cluster setup.